software/base/src/main/java/org/apache/brooklyn/entity/software/base/VanillaSoftwareProcessSshDriver.java

```diff
@@ -44,13 +47,20 @@ public VanillaSoftwareProcessSshDriver(EntityLocal entity, SshMachineLocation ma
 
     String downloadedFilename = null;
 
-    /** needed because the download url might be different! */
+    /**
+     * Needed because the download url and install commands are likely different for different VanillaSoftwareProcesses!
+     * This is particularly true for YAML entities. We take a hash of the download_url, instal_command and environment variables.
+     * We thus assume any templating of the script has already been done by this point.
+     */
     @Override
     protected String getInstallLabelExtraSalt() {
+        
         Maybe<Object> url = getEntity().getConfigRaw(SoftwareProcess.DOWNLOAD_URL, true);
-        if (url.isAbsent()) return null;
+        String installCommand = getEntity().getConfig(VanillaSoftwareProcess.INSTALL_COMMAND);
+        Map<String, Object> shellEnv = getEntity().getConfig(VanillaSoftwareProcess.SHELL_ENVIRONMENT);
+        
         // TODO a user-friendly hash would be nice, but tricky since we don't want it to be too long or contain path chars
```

Comment:
The hash is just a 4 (?) character hash generated from these fields, so does not look user-friendly when someone lists the contents of the `apps` directory. I'll leave this comment in for now.